barter-data/src/exchange/bybit/book/l1.rs

```diff
@@ -0,0 +1,48 @@
+use barter_instrument::exchange::ExchangeId;
+use chrono::Utc;
+
+use crate::{
+    event::{MarketEvent, MarketIter},
+    exchange::bybit::message::BybitMessage,
+    subscription::book::OrderBookL1,
+};
+
+use super::BybitOrderBookMessage;
+
+impl<InstrumentKey> From<(ExchangeId, InstrumentKey, BybitMessage)>
+    for MarketIter<InstrumentKey, OrderBookL1>
+where
+    InstrumentKey: Clone,
+{
+    fn from((exchange_id, instrument, message): (ExchangeId, InstrumentKey, BybitMessage)) -> Self {
+        match message {
+            BybitMessage::Orderbook(book) => Self::from((exchange_id, instrument, book)),
+            _ => Self(vec![]),
+        }
+    }
+}
+
+impl<InstrumentKey> From<(ExchangeId, InstrumentKey, BybitOrderBookMessage)>
+    for MarketIter<InstrumentKey, OrderBookL1>
+where
+    InstrumentKey: Clone,
+{
+    fn from(
+        (exchange, instrument, book): (ExchangeId, InstrumentKey, BybitOrderBookMessage),
+    ) -> Self {
+        let best_ask = book.data.asks.first().copied().map(Into::into);
+        let best_bid = book.data.bids.first().copied().map(Into::into);
```

Comment:
Nit pick: use `From` for readability (for example, in MRs), unless it's a generic bound constraint.